### ISSUE TITLE
Upload fails with 200 OK but error failed to fetch

### ISSUE THREAD

**Issue by lawrencenika** [ryanto/next-s3-upload#161]
this is my code: https://github.com/ryanto/next-s3-upload/issues/47#issuecomment-1741649478

in vercel it is showing 200OK but when I catch the error it shows `failed to fetch`

**Comment by quacrobat** [comment]
same, some images always upload, others always fail with this error.

**Comment by toklok** [comment]
Same here, looks like an error `fetch-http-handler`  LOC 56, any ideas how to fix this?

**Comment by dschlabach** [comment]
Can you explain how you were able to do this? In my package.json I have:

```
"@smithy/fetch-http-handler": "^2.2.2",
"@aws-sdk/client-s3": "3.317.0"
```

But my request is stuck on `pending` and I get this error:

```
TypeError: Failed to fetch
    at FetchHttpHandler.handle (webpack-internal:///(:3000/app-pages-browser)/./node_modules/next-s3-upload/node_modules/@aws-sdk/client-s3/node_modules/@smithy/fetch-http-handler/dist-es/fetch-http-handler.js:64:13)
    at async eval (webpack-internal:///(:3000/app-pages-browser)/./node_modules/@smithy/middleware-serde/dist-es/deserializerMiddleware.js:6:26)
    ```

**Comment by eggsy84** [comment]
Hey @dschlabach (and anyone else if getting this issue)

You can use the overrides part of NPM to temporarily fix this.

In the package.json if you put in

```
 "overrides": {
    "next-s3-upload": {
      "@aws-sdk/client-s3": "3.317.0"
    }
  },
```

It'll override the AWS SDK transitive dependency for next-s3-upload to be a specific version.

For context this is my overall package.json

```
{
  "name": "some-app",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "overrides": {
    "next-s3-upload": {
      "@aws-sdk/client-s3": "3.370.0"
    }
  },
  "dependencies": {
    "next": "13.5.4",
    "next-s3-upload": "^0.3.2",
    "react": "^18",
    "react-dom": "^18"
  },
  "devDependencies": {
    "autoprefixer": "^10",
    "eslint": "^8",
    "eslint-config-next": "13.5.4",
    "postcss": "^8",
    "tailwindcss": "^3"
  }
}
```

I went with version 3.370.0 because it had the most downloads and seemed to be working fine. It also didn't report any security vulnerabilities.

**Note:** I'd probably make a mental note to keep an eye on releases of s3-upload and once fixed you could remove this workaround

**Comment by ryanto** [comment]
@eggsy84 Awesome! thank you for debugging this one!

I can release a new version that pins 3.370.0, let me know what you think?

**Comment by eggsy84** [comment]
@ryanto No worries 🙌

Yeah could be a good way to go, if we knew how long until AWS fixed the issue in the SDK then you'd be able to call whether pinning the version is best way or whether you can continue to use the latest version. 

I'd hope they fix it soon as will be affecting others

**Comment by ryanto** [comment]
@eggsy84 @toklok thanks for much for debugging this! @toklok I just saw your edit and that helped clear things up!

I released a new version of next-s3-upload (`0.3.3`) that should have this all fixed. I added `@smithy/fetch-http-handler` as a dependency and wired it up with S3Client so everything just works as expected now.

Here's the AWS issue about this bug: https://github.com/aws/aws-sdk-js-v3/issues/5334

If anyone upgrades to `0.3.3` let me know how it goes!

Thanks again for all the help debugging this!

**Comment by quacrobat** [comment]
what about this fix? 
https://github.com/aws/aws-sdk-js-v3/issues/5334#issuecomment-1753486306

**Comment by ryanto** [comment]
@quacrobat that's the fix we're using in `0.3.3` :)

**Comment by quacrobat** [comment]
i can confirm 0.3.3 solves my problem.
